[CI] ReindexDataStreamTransportActionIT testAlreadyUpToDateDataStream failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #7293 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/p44rdxem57gk4)
- [elasticsearch-pull-request #65123 / part-2](https://gradle-enterprise.elastic.co/s/kd2rmub5jj2ca)
- [elasticsearch-periodic #7264 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/qn3b4roy3ekji)
- [elasticsearch-periodic #7264 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/w56673uaqzpfw)
- [elasticsearch-periodic-platform-support #7279 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/7q2j5ke7q75x2)
- [elasticsearch-periodic-platform-support #7279 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/ezjp43f5zjc2i)
- [elasticsearch-periodic-platform-support #7272 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/p5wu5uwwmdere)
- [elasticsearch-periodic-platform-support #7272 / ubuntu-2404-aarch64_checkpart2_platform-support-arm](https://gradle-enterprise.elastic.co/s/522sy3lnuzv3k)
- [elasticsearch-intake #20830 / part2](https://gradle-enterprise.elastic.co/s/rgskggo2waaci)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:migrate:internalClusterTest" --tests "org.elasticsearch.xpack.migrate.action.ReindexDataStreamTransportActionIT.testAlreadyUpToDateDataStream" -Dtests.seed=21AEE223CD422160 -Dtests.locale=asa-TZ -Dtests.timezone=Portugal -Druntime.java=24
```

**Applicable branches:**
9.0

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.migrate.action.ReindexDataStreamTransportActionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testAlreadyUpToDateDataStream),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <true>
     but: was <false>
```

**Issue Reasons:**
- [9.0] 9 failures in test testAlreadyUpToDateDataStream (8.1% fail rate in 111 executions)
- [9.0] 2 failures in step oraclelinux-8_platform-support-unix (66.7% fail rate in 3 executions)
- [9.0] 3 failures in pipeline elasticsearch-periodic-platform-support (75.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -338,9 +338,6 @@ tests:
 - class: org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT
   method: testSearchableSnapshotsInHotPhasePinnedToHotNodes
   issue: https://github.com/elastic/elasticsearch/issues/125683
-- class: org.elasticsearch.xpack.migrate.action.ReindexDataStreamTransportActionIT
-  method: testAlreadyUpToDateDataStream
-  issue: https://github.com/elastic/elasticsearch/issues/125727
 - class: org.elasticsearch.xpack.esql.spatial.SpatialExtentAggregationNoLicenseIT
   method: testStExtentAggregationWithPoints
   issue: https://github.com/elastic/elasticsearch/issues/125735

--- a/x-pack/plugin/migrate/src/internalClusterTest/java/org/elasticsearch/xpack/migrate/action/ReindexDataStreamTransportActionIT.java
+++ b/x-pack/plugin/migrate/src/internalClusterTest/java/org/elasticsearch/xpack/migrate/action/ReindexDataStreamTransportActionIT.java
@@ -69,10 +69,7 @@ public class ReindexDataStreamTransportActionIT extends ESIntegTestCase {
             dataStreamName
         );
         final int backingIndexCount = createDataStream(dataStreamName);
-        AcknowledgedResponse response = client().execute(
-            new ActionType<AcknowledgedResponse>(ReindexDataStreamAction.NAME),
-            reindexDataStreamRequest
-        ).actionGet();
+        client().execute(new ActionType<AcknowledgedResponse>(ReindexDataStreamAction.NAME), reindexDataStreamRequest).actionGet();
         String persistentTaskId = "reindex-data-stream-" + dataStreamName;
         AtomicReference<ReindexDataStreamTask> runningTask = new AtomicReference<>();
         for (TransportService transportService : internalCluster().getInstances(TransportService.class)) {
@@ -91,12 +88,14 @@ public class ReindexDataStreamTransportActionIT extends ESIntegTestCase {
             );
         }
         ReindexDataStreamTask task = runningTask.get();
-        assertNotNull(task);
-        assertThat(task.getStatus().complete(), equalTo(true));
-        assertNull(task.getStatus().exception());
-        assertThat(task.getStatus().pending(), equalTo(0));
-        assertThat(task.getStatus().inProgress(), equalTo(Set.of()));
-        assertThat(task.getStatus().errors().size(), equalTo(0));
+        assertBusy(() -> {
+            assertNotNull(task);
+            assertThat(task.getStatus().complete(), equalTo(true));
+            assertNull(task.getStatus().exception());
+            assertThat(task.getStatus().pending(), equalTo(0));
+            assertThat(task.getStatus().inProgress(), equalTo(Set.of()));
+            assertThat(task.getStatus().errors().size(), equalTo(0));
+        });
 
         assertBusy(() -> {
             GetMigrationReindexStatusAction.Response statusResponse = client().execute(

